src/main/java/uk/gov/digital/ho/hocs/audit/export/adapter/DateAdapter.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.digital.ho.hocs.audit.export.adapter;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static uk.gov.digital.ho.hocs.audit.export.infoclient.ExportViewConstants.*;
+import static uk.gov.digital.ho.hocs.audit.export.infoclient.ExportViewConstants.DATE_SECTION.*;
+
+
+@Slf4j
+public class DateAdapter implements ExportViewFieldAdapter {
+
+    @Override
+    public String getAdapterType() {
+        return FIELD_ADAPTER_DATE;
+    }
+
+    /**
+     * The aim of this function is to identify dates that are malformed by having a zero in front of the
+     * year, month or day (0yyyy-0mm-0dd). The REGEX used will identify any combination of the zero leading digits.
+     * Once identified, this function converts the date to a format without the leading zero (yyyy-mm-dd).
+     *
+     * @param input The ExportViewFieldAdapter interface supports any kind of object but the calling code only ever passes a string.
+     * @return A string value of the converted (corrected) date or the original value if it is not a date.
+     */
+    @Override
+    public String convert(Object input) {
+        if (input instanceof String) {
+            String date = (String)input;
+            if (date.matches(MALFORMED_DATE_REGEX)) {
+                String year = retrieveSection(YEAR, date);
```

Comment:
High: Safer to use groups in the MALFORMED_DATE_REGEX and then extract matched parts from that instead of retrieving sections of the date again. regexes for sections could easily be broken.

Reply:
This operations is for an extract, which is applied to all fields as we do not have an implementation for the types of data.
In essence we cannot match groups as it will then match partial data from other types like text fields.
It has to be all or nothing on the exact error we are looking for, ie, [0]yyyy-[0]mm-[0]dd

Reply:
Code changed to use find on compiled pattern.